tst/com/amazon/corretto/crypto/provider/test/EvpKeyFactoryTest.java

```diff
@@ -87,10 +87,12 @@ public static void setupParameters() throws Exception {
 
     for (String algorithm : ALGORITHMS) {
       KeyPairGenerator kpg;
-      if (algorithm.startsWith("ML-DSA")) {
+      if (algorithm.startsWith("ML-DSA")
+          || (algorithm.startsWith("Ed") && TestUtil.getJavaVersion() < 15)) {
```

Comment:
Bit of code smell here, I'm paranoid that matching on startsWith("Ed") is a bit too general.

Reply:
This is very fair. Since it's a maintainability issue and not a correctness issue, I'll do a follow-up PR to match `algorithm` more precisely.